pytest_splunk_addon/standard_lib/cim_tests/test_generator.py

```diff
@@ -252,3 +257,25 @@ def generate_mapped_datamodel_tests(self):
             {"eventtypes": eventtypes},
             id=f"mapped_datamodel_tests",
         )
+
+    def generate_recommended_fields_tests(self, tokenized_events):
+        for event in tokenized_events:
+            if not event.requirement_test_data:
+                continue
+            for _, datamodel in event.requirement_test_data["datamodels"].items():
+                model, *datasets = datamodel.split(":")
+                model = model.replace(" ", "_")
+                if datasets:
+                    datasets = [dataset.replace(" ", "_") for dataset in datasets]
+
+                fields = (
+                    list(event.requirement_test_data["cim_fields"].keys())
+                    + event.requirement_test_data["missing_recommended_fields"]
+                )
+                for exception in event.requirement_test_data["exceptions"]:
+                    fields.append(exception["name"])
+
+                yield pytest.param(
+                    {"datamodel": model, "datasets": datasets, "fields": fields},
+                    id=f"{model}-{'-'.join(datasets)}::sample_name::{event.sample_name}::host::{event.metadata.get('host')}",
```

Comment:
Do we need to generate test with a `host` name?

I am asking because I would like to remove `host` as a required option from the req tests.

Reply:
the host here is added only for id parameter so basically as a description to test. Generally, as I see it now host field is being used in PSA as a unique field value because a number is being added as a suffix to all hosts

Reply:
host is not used in this test logic